src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/PreferHashDataOverComputeHashTests.cs

```diff
@@ -0,0 +1,624 @@
+                .WithArguments("System.Security.Cryptography.SHA256")
+                .WithLocation(0)
+                .WithLocation(1)
+                .WithLocation(2));
+        }
+
+        [Fact]
+        public async Task BasicObjectCreationUsingBlockCase()
+        {
+            string vbInput = @"
+Imports System
+Imports System.Security.Cryptography
+
+Public Class Test
+    Public Shared Sub TestMethod()
+        Dim buffer = New Byte(1023) {}
+        {|#2:Using sha256 As New SHA256Managed()|}
+            Dim digest As Byte() = {|#0:sha256.ComputeHash({|#1:buffer|})|}
+        End Using
+    End Sub
+End Class
+";
+
+            string vbFix = @"
+Imports System
+Imports System.Security.Cryptography
+
+Public Class Test
+    Public Shared Sub TestMethod()
+        Dim buffer = New Byte(1023) {}
+        Dim digest As Byte() = SHA256.HashData(buffer)
+    End Sub
+End Class
+";
+            await TestVB(
+                vbInput,
+                vbFix,
+                VerifyCS.Diagnostic(PreferHashDataOverComputeHashAnalyzer.StringRule)
+                .WithArguments("System.Security.Cryptography.SHA256")
+                .WithLocation(0)
+                .WithLocation(1)
+                .WithLocation(2));
+
+        }
+
+        [Fact]
+        public async Task BasicCreateHelperUsingBlockCase()
+        {
+            string vbInput = @"
+Imports System
+Imports System.Security.Cryptography
+
+Public Class Test
+    Public Shared Sub TestMethod()
+        Dim buffer = New Byte(1023) {}
+        {|#2:Using sha256 As SHA256 = SHA256.Create()|}
+            Dim digest As Byte() = {|#0:sha256.ComputeHash({|#1:buffer|})|}
+        End Using
+    End Sub
+End Class
+";
+
+            string vbFix = @"
+Imports System
+Imports System.Security.Cryptography
+
+Public Class Test
+    Public Shared Sub TestMethod()
+        Dim buffer = New Byte(1023) {}
+        Dim digest As Byte() = SHA256.HashData(buffer)
+    End Sub
+End Class
+";
+            await TestVB(
+                vbInput,
+                vbFix,
+                VerifyCS.Diagnostic(PreferHashDataOverComputeHashAnalyzer.StringRule)
+                .WithArguments("System.Security.Cryptography.SHA256")
+                .WithLocation(0)
+                .WithLocation(1)
+                .WithLocation(2));
+
+        }
+
+        private static Task TestCS(string source, string corrected, params DiagnosticResult[] expected)
+        {
+            var test = new VerifyCS.Test
+            {
+                TestCode = source,
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                LanguageVersion = LanguageVersion.Preview,
+                FixedCode = corrected,
+            };
+
+            test.ExpectedDiagnostics.AddRange(expected);
+            return test.RunAsync();
+        }
+
+        private static Task TestVB(string source, string corrected, params DiagnosticResult[] expected)
```

Comment:
There are only two VB tests. Can you please add the same tests in both C# and VB? Ideally, in two separate files.

Reply:
I created more VB Tests. There's 1 case in VB that does not work. 
When you call `SHA256.Create()` to a `HashAlgorithm` variable, the subclass `SHA256` information is lost in VB.